lib/src/base-styles/_nx-btn.scss

```diff
@@ -6,48 +6,75 @@
  */
 @use '../scss-shared/nx-container-helpers';
 @use '../scss-shared/nx-text-helpers';
+@use '../scss-shared/nx-dark-mode-helpers';
 
-%button-focus-outline {
+%transparent-button-focus-outline {
+  outline-offset: calc(0px - var(--nx-border-width-button) - 4px);
+  outline: 2px solid;
+}
+
+%solid-button-focus-outline {
   outline-offset: -5px;
   outline: 2px solid;
```

Comment:
Consider that the CSS generated by these placeholder classes ends up like this: 
![image](https://user-images.githubusercontent.com/3630091/229905518-25f88276-7282-4166-bfea-b4b5285a08e4.png)

There's actually no point in including the `outline` property in a placeholder or mixin because it gets applied to `nx-btn` itself, a class which all more specialized buttons should also have.  That style can just be right in `.nx-btn:focus:not(active)`.  Then if that is removed from these placeholders, also consider that `solid-button-focus-outline` is only used in one place, so the `outline-offset` can just be moved to that place and that placeholder class can be removed.  `transparent-button-focus-outline` still has some value in reuse, but I think a placeholder class is probably more complex than necessary for it at that point.  I'd just use a scss variable to store the calc expression.

Reply:
> There's actually no point in including the `outline` property in a placeholder or mixin because it gets applied to `nx-btn` itself, a class which all more specialized buttons should also have. That style can just be right in `.nx-btn:focus:not(active)`. Then if that is removed from these placeholders, also consider that `solid-button-focus-outline` is only used in one place, so the `outline-offset` can just be moved to that place and that placeholder class can be removed. `transparent-button-focus-outline` still has some value in reuse, but I think a placeholder class is probably more complex than necessary for it at that point. I'd just use a scss variable to store the calc expression instead of a placeholder there.

Great, that makes a lot of sense! Cleaned up here: [02ff31b](https://github.com/sonatype/sonatype-react-shared-components/pull/1038/commits/02ff31b5d7d3b5f482891c51ed331880c29238d5)